### PR TITLE
Modified corpus_viz as suggested by the team

### PR DESCRIPTION
**Pull request by ssyayayy**

### CODE REVIEW

File: corpysprofiling/corpysprofiling.py [hunk 1]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
-    >>> plt.imshow(wordcloud, interpolation="bilinear")
-    >>> plt.axis("off")
     >>> corpysprofiling.corpus_viz("How many species of animals are there in Russia?")['df used for bar']
-    >>> df.plot.bar(rot=0, x='words')
-    >>> plt.xticks(rotation=90)
-    >>> plt.xlabel("Words")
-    >>> corpysprofiling.corpus_viz("How many species of animals are there in Russia?", 15)
-    TypeError: Input must be a string
+    >>> corpysprofiling.corpus_viz("How many species of animals are there in Russia?")['word length bar chart']
+
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:117]: Generated wordcloud looks very nice!

File: corpysprofiling/corpysprofiling.py [hunk 2]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
+    plt.bar(x=df['words'], height = df['length'])
     plt.xticks(rotation=90)
-    plt.xlabel("Words")
-    
-    if display==False:
-        plt.close()
-        plt.close(1)
+    plt.xlabel('Words')
+    plt.title('Length for the Most Common Words')
+    plt.close(1)
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:149]: I liked your idea of including word frequencies 

```suggestion
    bar_freq = (alt.Chart(df).encode(
        x=alt.X("word", sort='-y', title="Word"), 
        y=alt.Y("count()", title="Frequency")).mark_bar()
    .properties(title="Frequency of Words"))
```

File: corpysprofiling/corpysprofiling.py [hunk 3]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
     
-    return {'word cloud': wordcloud, 'df used for bar': df}
+    return {'word cloud': wordcloud_fig, 
+            'df used for bar': df, 
+            "word length bar chart":bar_fig}
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:152]:
```suggestion
            "word length bar chart":bar_length}
```

File: corpysprofiling/corpysprofiling.py [hunk 4]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
-    if display==False:
-        plt.close()
-        plt.close(1)
+    plt.xlabel('Words')
+    plt.title('Length for the Most Common Words')
+    plt.close(1)
     
-    return {'word cloud': wordcloud, 'df used for bar': df}
+    return {'word cloud': wordcloud_fig, 
+            'df used for bar': df,
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:151]:
```suggestion
            "word freq bar chart": bar_freq, 
```

File: corpysprofiling/corpysprofiling.py [hunk 5]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
     """
     return
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:94]:
```suggestion
import altair as alt
```

File: corpysprofiling/corpysprofiling.py [hunk 6]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
     -------
     dictionary
-        contains a wordcloud.WordCloud, which can be used to present a word cloud,
-        and a data frame, which can be used to draw a bar chart for words and word lengths
-
-    Raises
-    -------
-    TypeError
-        If argument passed is of wrong type
+        contains a word cloud, a data frame, which can be used to draw a bar chart for words and word lengths,
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:107]:
```suggestion
        contains a word cloud, a histogram of word length frequencies, and a histogram of word frequencies 
```

File: corpysprofiling/corpysprofiling.py [hunk 7]
```diff
@@ -92,71 +92,64 @@ def corpus_analysis(corpus):
...
     dictionary
-        contains a wordcloud.WordCloud, which can be used to present a word cloud,
-        and a data frame, which can be used to draw a bar chart for words and word lengths
-
-    Raises
-    -------
-    TypeError
-        If argument passed is of wrong type
+        contains a word cloud, a data frame, which can be used to draw a bar chart for words and word lengths, 
+        and the bar chart
```
Comment (jkim222383) [inline corpysprofiling/corpysprofiling.py:108]:
```suggestion

```

File: tests/test_corpysprofiling.py [hunk 8]
```diff
@@ -14,13 +16,17 @@ def test_corpus_viz():
...
 
     # Test whether the corpus_viz returns a dictionary
-    assert isinstance(corpus_viz(corpus1, display=False), dict), "Return type is not a dict"
+    assert isinstance(corpus_viz(corpus1), dict), "Return type is not a dict"
+
     # Test whether user can access the data frame used to plot the bar chart
-    assert isinstance(corpus_viz(corpus1, display=False)['df used for bar'], pd.DataFrame), "Return type is not a data frame"
-    # Test whether user can access the word cloud object
-    assert isinstance(corpus_viz(corpus2, display=False)['word cloud'], WordCloud), "Return type is not a word cloud"
+    assert isinstance(corpus_viz(corpus1)['df used for bar'], pd.DataFrame), "Return type is not a data frame"
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:22]:
```suggestion
    assert isinstance(corpysprofiling.corpus_viz(corpus1)['word length bar chart'], alt.Chart), "Return type is not an altair Chart"
```

File: tests/test_corpysprofiling.py [hunk 9]
```diff
@@ -14,13 +16,17 @@ def test_corpus_viz():
     corpus2 = "Let's assume that this unsupervised model is used to assist human experts to identify fraud transaction. So instead of making humans examine all 284,807 transactions for fraud transactions, this model would extract transactions which look suspicious and pass them to humans for examination. So our goal is to come up with a list of transactions which look suspicious.We said before that PCA won't be able to capture characteristic features of fraud transactions because they are like outliers (occur very rarely) in our dataset, and so the reconstruction error would be higher for them compared to non-fraud transactions. But what do we mean by high reconstruction error? What should be threshold which makes a transaction suspicious?"
 
     # Test whether the corpus_viz returns a dictionary
-    assert isinstance(corpus_viz(corpus1, display=False), dict), "Return type is not a dict"
+    assert isinstance(corpus_viz(corpus1), dict), "Return type is not a dict"
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:19]:
```suggestion
    assert isinstance(corpysprofiling.corpus_viz(corpus1), dict), "Return type is not a dict"
```

File: tests/test_corpysprofiling.py [hunk 10]
```diff
@@ -14,13 +16,17 @@ def test_corpus_viz():
...
+
     # Test whether user can access the data frame used to plot the bar chart
-    assert isinstance(corpus_viz(corpus1, display=False)['df used for bar'], pd.DataFrame), "Return type is not a data frame"
-    # Test whether user can access the word cloud object
-    assert isinstance(corpus_viz(corpus2, display=False)['word cloud'], WordCloud), "Return type is not a word cloud"
+    assert isinstance(corpus_viz(corpus1)['df used for bar'], pd.DataFrame), "Return type is not a data frame"
+
+    # Test whether user can access the word cloud 
+    assert isinstance(corpus_viz(corpus2)['word cloud'], plt.Figure), "Return type is not a word cloud"
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:25]:
```suggestion
    assert isinstance(corpysprofiling.corpus_viz(corpus2)['word cloud'], plt.Figure), "Return type is not a word cloud"
```

File: tests/test_corpysprofiling.py [hunk 11]
```diff
@@ -14,13 +16,17 @@ def test_corpus_viz():
...
+
     # Test whether if else statement works - the function should only consider the top 30 most frequently used words
-    assert corpus_viz(corpus2, display=False)['df used for bar'].shape[0] <= 30, "Too many words"
+    assert corpus_viz(corpus1)['df used for bar'].shape[0] == 5, "Too many or too few words"
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:28]:
```suggestion
        assert isinstance(corpysprofiling.corpus_viz(corpus1)['word freq bar chart'], alt.Chart), "Word frequency histogram is not an altair Chart"
        assert corpysprofiling.corpus_viz(corpus1)['word freq bar chart'].data.shape[0] == 5, "Number of words does not match number of unique tokens in corpus."
```

File: tests/test_corpysprofiling.py [hunk 12]
```diff
@@ -14,13 +16,17 @@ def test_corpus_viz():
...
+    assert corpus_viz(corpus2)['df used for bar'].shape[0] <= 30, "Too many words"
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:29]:
```suggestion
    assert corpysprofiling.corpus_viz(corpus2)['word freq bar chart'].data.shape[0] <= 30, "Too many words"
```

File: tests/test_corpysprofiling.py [hunk 13]
```diff
@@ -3,6 +3,8 @@
 import numpy as np
 import pandas as pd
 from collections import Counter
+import matplotlib.pyplot as plt
+from wordcloud import WordCloud
```
Comment (jkim222383) [inline tests/test_corpysprofiling.py:8]:
```suggestion
import altair as alt
```

**Review by jkim222383 — CHANGES_REQUESTED** [review]
The word cloud generated looks really nice! I realized that the bar chart generated actually shows frequencies of words. I made suggestions to modify the function so it returns the correct frequencies of word lengths, but I kept your original idea of plotting histograms of word frequencies since I think that's also useful visualization.  

I realized that the need to return both `pd.DataFrame` and `plt` plot object can be avoided by using `altair.Chart`, since the DataFrame for the chart can be accessed by the user simply using `.data` attribute of the returned `altair.Chart` object.

**Review by ssyayayy — COMMENTED** [review]
Resolved the issues as suggested. Thanks a lot for your valuable comments!
If no further issues, you may go ahead and merge this to the main branch.

**Pull request merged**